Include/internal/pycore_floatobject.h

```diff
@@ -55,6 +55,8 @@ extern PyObject* _Py_string_to_number_with_underscores(
 
 extern double _Py_parse_inf_or_nan(const char *p, char **endptr);
 
+extern char * _Py_dg_dtoa_hex(double x, int precision, int always_add_sign,
```

Comment:
I suggest calling this something clearer like `_Py_float_to_hex`. The `dg_dtoa` stuff doesn't make much sense here: "dg" is a reference to David Gay, since he wrote the original version of the dtoa.c code that Python's dtoa.c is baed on; this code has nothing to do with Gay's code.

Reply:
Thanks, does make sense.

Initially that helper was a static function in pystrtod.c; just as ``_Py_dg_dtoa()`` it, well, "convert double to ASCII string"...  This was exported and moved to floatobject.c to (1) slighly reduce diff and (2) support workaround for hex().

I think (1) goal wasn't very successful.  Then, maybe this helper belongs to pystrtod.c?  If so, ``_Py_dtoa_hex()`` sounds ok?